packages/core/saved-objects/core-saved-objects-migration-server-internal/src/model/model.ts

```diff
@@ -736,6 +736,15 @@ export const model = (currentState: State, resW: ResponseType<AllActionStates>):
         ...stateP,
         controlState: 'CALCULATE_EXCLUDE_FILTERS',
       };
+    } else if (isTypeof(res.left, 'source_equals_target')) {
+      // As part of a reindex-migration, we wanted to block the source index to prevent updates
+      // However, this migrator's source index matches the target index.
+      // Thus, another instance's migrator is ahead of us. We simply abort and retry
+      return {
+        ...stateP,
+        controlState: 'FATAL',
+        reason: `The ${stateP.currentAlias} migrator is attempting to lock the [${res.left.index}] index. Another migrator is likely ahead of us in the process, aborting.`,
+      };
```

Comment:
Thinking out loud: Are we really forced to terminate with a FATAL here? Is there a risk if we were to just skip the migration and succeed? I guess the risk could be to finish and be 'ready' because the node completing the migration is?

Thinking of the test that helped identify this problem - having the node terminate with a FATAL will still be considered as a failure, and ideally we would be resillient to this instead of FATAL-ing.

But if that's the safest option, it's fine.

Reply:
That's a good point. If the current version index exists, it surely means there's another migrator performing a migration, which created that index.

I am not comfortable with jumping straight to `DONE`, as the other migrator might still be in the process of updating the target index (e.g. putting _mappings), but perhaps we could jump to `REFRESH_TARGET` (step after creating that index), so that we resume the flow at a point where we still perform the rest of the steps of the _reindex path_.

WDYT?

Reply:
Updated in https://github.com/elastic/kibana/pull/185939/commits/666637714a7a7fe2357a9422b5781a76fbcaaf2b

Reply:
Yeah, I think it's better that way (and I can't think of any later-stage race this could cause - even if it doesn't mean there can't be 😄)